hadoop-hdfs-project/hadoop-hdfs/src/main/resources/hdfs-default.xml

```diff
@@ -3003,10 +3003,10 @@
 </property>
 
 <property>
-  <name>dfs.datanode.fsdatasetasyncdisk.max.threads.per.volume</name>
-  <value>4</value>
+  <name>dfs.datanode.fsdatasetasyncdisk.core.threads.per.volume</name>
+  <value>1</value>
   <description>
-    The maximum number of threads per volume used to process async disk
+    The number of core threads per volume used to process async disk
```

Comment:
Suggest you leave the old config alone and introduce a new one for core thread count.

Reply:
@saintstack Thanks for your comment.

> During bulk delete, the executor did not blossom up to the max number of threads per volume? Upping the max wouldn't help w/ your case?

The threadPoolExecutor used one unbounded LinkedBlockingQueue, so the executor will not blossom up to the max number of threads per volume, and upping the max wouldn't help this case.


Because the threadPoolExecutor use one unbounded LinkedBlockingQueue, so I want to set the maximum number of threads same with the number of core threads.

Do you have any good suggestions?